### PR TITLE
feat(medusa): Filtering Customer Orders

### PR DESCRIPTION
**Pull request by pKorsholm**
**What** 
- add filtering to `/customers/me/orders` on the following parameters: 
  - `id?: string;`
  - `q?: string;`
  - `status?: OrderStatus[];`
  - `fulfillment_status?: FulfillmentStatus[];`
  - `payment_status?: PaymentStatus[];`
  - `display_id?: string;`
  - `cart_id?: string;`
  - `email?: string;`
  - `region_id?: string;`
  - `currency_code?: string;`
  - `tax_rate?: string;`
  - `created_at?: DateComparisonOperator;`
  - `updated_at?: DateComparisonOperator;`
  - `canceled_at?: DateComparisonOperator;`

**How**
- Following the standard approach across other filtering endpoints

**Testing** 
- Add the following integration tests: 
  - `looks up completed orders`
  - `looks up cancelled and completed orders`

### CODE REVIEW

File: packages/medusa/src/api/routes/store/customers/list-orders.ts [hunk 1]
```diff
@@ -79,12 +104,22 @@ export default async (req, res) => {
...
-  const [orders, count] = await orderService.listAndCount(selector, listConfig)
+  const filterableFields = _.omit(validated, [
+    "limit",
+    "offset",
+    "expand",
+    "fields",
+  ])
+
+  const [orders, count] = await orderService.listAndCount(
+    _.pickBy(filterableFields, identity),
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/customers/list-orders.ts:115]:
```suggestion
    _.pickBy(filterableFields, (val) => typeof val !== `undefined`),
```

File: packages/medusa/src/api/routes/store/customers/list-orders.ts [hunk 2]
```diff
@@ -1,6 +1,19 @@
 import { Type } from "class-transformer"
-import { IsNumber, IsOptional, IsString } from "class-validator"
+import {
+  IsEnum,
+  IsNumber,
+  IsOptional,
+  IsString,
+  ValidateNested,
+} from "class-validator"
+import _, { identity } from "lodash"
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/customers/list-orders.ts:9]: nit (not much difference really, so take it or leave it 😅 ):
```suggestion
import { identity, omit, pickBy } from "lodash"
```

**Review by olivermrbl — APPROVED** [review]
LGTM! Great 👍  Just a couple of comments.

File: packages/medusa/src/api/routes/store/customers/index.ts [hunk 3]
```diff
@@ -34,7 +34,11 @@ export default (app, container) => {
   route.get("/me", middlewares.wrap(require("./get-customer").default))
   route.post("/me", middlewares.wrap(require("./update-customer").default))
 
-  route.get("/me/orders", middlewares.wrap(require("./list-orders").default))
+  route.get(
+    "/me/orders",
+    middlewares.normalizeQuery(),
```
Comment (adrien2p) [inline packages/medusa/src/api/routes/store/customers/index.ts:39]: suggestion: we could defenatly integrate the new middleware approach here and in the handler. wdyt?

File: packages/medusa/src/api/routes/store/customers/list-orders.ts [hunk 4]
```diff
@@ -79,12 +104,22 @@ export default async (req, res) => {
...
+  const [orders, count] = await orderService.listAndCount(
+    _.pickBy(filterableFields, identity),
+    listConfig
+  )
 
   res.json({ orders, count, offset: validated.offset, limit: validated.limit })
 }
 
-export class StoreGetCustomersCustomerOrdersParams {
+export class StoreGetCustomersCustomerOrdersPaginationParams {
```
Comment (adrien2p) [inline packages/medusa/src/api/routes/store/customers/list-orders.ts:122]: suggestion: can extends `FindParams`

**Review by adrien2p — APPROVED** [review]
LGTM 🎉 only two small suggestions

**Pull request merged**